Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -77,149 +78,207 @@ extension NIOClientTCPBootstrap {
         on eventLoop: EventLoop,
         host: String,
         requiresTLS: Bool,
-        configuration: HTTPClient.Configuration
-    ) throws -> NIOClientTCPBootstrap {
-        var bootstrap: NIOClientTCPBootstrap
+        configuration: HTTPClient.Configuration,
+        sslContextCache: SSLContextCache,
+        logger: Logger
+    ) -> EventLoopFuture<NIOClientTCPBootstrap> {
+        let bootstrap: EventLoopFuture<NIOClientTCPBootstrap>
         #if canImport(Network)
             // if eventLoop is compatible with NIOTransportServices create a NIOTSConnectionBootstrap
             if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *), let tsBootstrap = NIOTSConnectionBootstrap(validatingGroup: eventLoop) {
                 // if there is a proxy don't create TLS provider as it will be added at a later point
                 if configuration.proxy != nil {
-                    bootstrap = NIOClientTCPBootstrap(tsBootstrap, tls: NIOInsecureNoTLS())
+                    bootstrap = eventLoop.makeSucceededFuture(NIOClientTCPBootstrap(tsBootstrap,
+                                                                                    tls: NIOInsecureNoTLS()))
                 } else {
                     // create NIOClientTCPBootstrap with NIOTS TLS provider
                     let tlsConfiguration = configuration.tlsConfiguration ?? TLSConfiguration.forClient()
                     let parameters = tlsConfiguration.getNWProtocolTLSOptions()
                     let tlsProvider = NIOTSClientTLSProvider(tlsOptions: parameters)
-                    bootstrap = NIOClientTCPBootstrap(tsBootstrap, tls: tlsProvider)
+                    bootstrap = eventLoop.makeSucceededFuture(NIOClientTCPBootstrap(tsBootstrap, tls: tlsProvider))
                 }
             } else if let clientBootstrap = ClientBootstrap(validatingGroup: eventLoop) {
-                bootstrap = try clientBootstrap.makeClientTCPBootstrap(host: host, requiresTLS: requiresTLS, configuration: configuration)
+                bootstrap = sslContextCache.sslContext(tlsConfiguration: configuration.tlsConfiguration ?? .forClient(),
+                                                       eventLoop: eventLoop,
+                                                       logger: logger)
```

Comment:
This is hit with in a test that doesn't require `https`: `testUploadStreamingBackpressure`

Reply:
great catch! Thanks you